packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/entity_type/read.rs

```diff
@@ -0,0 +1,142 @@
+use std::str::FromStr;
+
+use async_trait::async_trait;
+use error_stack::{IntoReport, Result, ResultExt};
+use futures::{stream, StreamExt, TryStreamExt};
+use type_system::{uri::VersionedUri, EntityType};
+
+use crate::store::{
+    postgres::resolve::{PostgresContext, Record},
+    query::{
+        Literal, PathSegment, Resolve, ResolveError, UNIMPLEMENTED_LITERAL_OBJECT,
+        UNIMPLEMENTED_WILDCARDS,
+    },
+};
+
+async fn resolve_entity_type(
+    uri: &VersionedUri,
+    path: &[PathSegment],
+    context: &(impl PostgresContext + Sync),
+) -> Result<Literal, ResolveError> {
+    context
+        .read_versioned_entity_type(uri)
+        .await
+        .change_context(ResolveError::StoreReadError)?
+        .resolve(path, context)
+        .await
+}
+
+#[async_trait]
+impl<C> Resolve<C> for Record<EntityType>
+where
+    C: PostgresContext + Sync,
+{
+    async fn resolve(&self, path: &[PathSegment], context: &C) -> Result<Literal, ResolveError> {
+        match path {
+            [] => todo!("{}", UNIMPLEMENTED_LITERAL_OBJECT),
+            [segment, segments @ ..] => {
+                // TODO: Avoid cloning on literals
+                //   see https://app.asana.com/0/0/1202884883200947/f
+                let literal = match segment.identifier.as_str() {
+                    "uri" => Literal::String(self.record.id().base_uri().to_string()),
+                    "version" => Literal::Version(self.record.id().version(), self.is_latest),
+                    "title" => Literal::String(self.record.title().to_owned()),
+                    "description" => self
+                        .record
+                        .description()
+                        .map_or(Literal::Null, |description| {
+                            Literal::String(description.to_owned())
+                        }),
+                    "default" | "examples" => todo!("{}", UNIMPLEMENTED_LITERAL_OBJECT),
+                    "properties" => match segments {
```

Comment:
Quick question, we're using the literal `"propertyTypes"` to get property types references on property types, and here we specify "properties". Is the decision to have `"properties"` here for being closer to the schema? 

I think this should be `"propertyTypes"` as well since we're not dealing with properties of entities but their type counterparts

Reply:
Yes, I used `"properties"` to match the schema. I think if we want to move to `"propertyTypes"` the schema should also be updated. cc @hashintel/backend thoughts?

Reply:
We spoke about this offline and agreed that we should stick with `"properties"`.